src/nvim/options.lua

```diff
@@ -311,6 +311,8 @@ return {
       cb = 'did_set_background',
       defaults = { if_true = 'dark' },
       values = { 'light', 'dark' },
+      -- TODO(lewis6991): #31944 removed the ability to set to empty. Is empty
+      -- allowed?
```

Comment:
As you probably noticed, `:let &bg=''` in both Nvim and Vim sets the option to its "default" but does not set the value itself. Very strange. Probably a bug. Not documented. So let's treat it in a "strict" way and not support it.

Same for the other cases: if it's not documented then it's probably an accident.